### ISSUE TITLE
GQL-modules injection problem with type-graphql if no params to the function

### ISSUE THREAD

**Issue by mohamedGamalAbuGalala** [Urigo/graphql-modules#1019]
**Describe the bug**
I work on a project with `GQL-modules` and `type-graphql` both works together like a charm but I'm having some problems when I create a query/mutation and the function doesn't need to take any input, as it gives me an error that the injected instance is undefined.

**Module code** 
```javascript

export const HopeModule = new GraphQLModule({
  extraSchemas: [
    buildSchemaSync({
      resolvers: [HopeQuery],
      container: ({ context }) => context.injector
    })
  ],
  providers: [
    HopeQuery,
    HopeProvider
  ]
})

```

**Resolver code**
```javascript
   
@Resolver()
export class HopeQuery {
  constructor (private readonly hopeProvider: HopeProvider) {}

  @Query(() => String)
  hopeQuery (): string {
    return this.hopeProvider.init() // should return "I'm happy :)"
  }
}

```
**Expected behavior**
When I call the hopeQuery it returns the "I'm happy :)" string from the provider.

**Actual Behavior**
It gives me "Cannot read property 'init' of undefined" which means that hopeProvider is undefined.
But replacing this `hopeQuery` method with another taking any parameter works!!
```javascript
  @Query(() => String)
  hopeQuery (@Ctx { me }: IMyContext): string {
    return this.hopeProvider.init() // should return "I'm happy :)"
  }
```
The last snippet will work!!

**Comment by mohamedGamalAbuGalala** [comment]
MichalLytek/type-graphql#588

**Comment by mohamedGamalAbuGalala** [comment]
Closing this after digging more in this lead me that the error was in some package or some piece of code affecting `reflect-metadata` package which blows everything.
But for referencing the error was in some changes in one file, when I discarded them it goes away.

Maybe related to connecting error with `neode` (related to adam-cowley/neode#56
 or adam-cowley/neode#14)or another error.

god-only-know-why the reason causing it.
